packages/salesforcedx-vscode-replay-debugger/src/states/variableAssignmentState.ts

```diff
@@ -34,13 +37,38 @@ export class VariableAssignmentState implements DebugLogState {
         ref = this.fields[5];
       }
       let container: ApexVariableContainer | undefined;
+      let map: Map<String, VariableContainer> | undefined;
       if (logContext.getStaticVariablesClassMap().has(className)) {
-        const statics = logContext.getStaticVariablesClassMap().get(className)!;
-        container = statics.get(name)! as ApexVariableContainer;
+        map = logContext.getStaticVariablesClassMap().get(className)!;
+        container = map.get(varName)! as ApexVariableContainer;
+        if (container.variablesRef !== 0 && !logContext.getRefsMap().has(ref)) {
+          container = new ApexVariableContainer(
+            container.name,
+            '',
+            container.type
+          );
+          map.set(varName, container);
+        }
       } else if (frameInfo.locals.has(varName)) {
         // if name does not contain '.' (i.e. this.attr or a.Name), it should be in locals and we can update the value
-        container = frameInfo.locals.get(varName) as ApexVariableContainer;
+        map = frameInfo.locals;
+        container = map.get(varName) as ApexVariableContainer;
+      }
+
+      if (
+        container &&
+        map &&
+        container.variablesRef !== 0 &&
+        !logContext.getRefsMap().has(ref)
+      ) {
```

Comment:
Without this check, the container for the old reference would be the value for the new ref in the map. This would later be overwritten with the values of the new ref which we don't want in case any other variables are still pointing to the old ref.